pkg/bootstrap/bootstrap.go

```diff
@@ -181,24 +180,22 @@ func extract(targetDir string, dirs map[string]string, reader io.Reader) error {
 			return err
 		}
 
-		targetName := filepath.Join(targetDir, h.Name)
-		targetDir := filepath.Dir(h.Name)
+		sourceDir := filepath.Dir(h.Name)
+		targetDir, ok := dirs[sourceDir]
+		if !ok {
+			continue
+		}
 
 		switch h.Typeflag {
 		case tar.TypeDir:
-			if _, ok := dirs[targetDir]; !ok {
-				continue
-			}
-			if _, err := os.Stat(targetName); err != nil {
-				if err := os.MkdirAll(targetName, 0755); err != nil {
+			if _, err := os.Stat(targetDir); err != nil {
```

Comment:
Do we care what the error is or just the existance?

Reply:
Yeah I'm not sure how defensive we need to be - if it's a permissions error or something else it'll bubble up when the mkdir fails?

Reply:
I hate to prolong this but for clarity of intent, it'd probably be good to perform an explicit existence check, make the directory if it doesn't, and log or return otherwise.

Reply:
Turns out MkdirAll will actually stat the target path for you and just returns without error if it already exists.

Reply:
Should we remove the stat call all together then?